From ac0fa764dfdd5e45ccda6f91c6e136e84e284c4a Mon Sep 17 00:00:00 2001
From: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date: Mon, 28 Aug 2017 13:59:12 -0400
Subject: [PATCH 195/206] Use grub-file to figure out whether multiboot2 should
 be used for Xen.gz

The multiboot2 is much more preferable than multiboot. Especiall
if booting under EFI where multiboot does not have the functionality
to pass ImageHandler.

Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
---
 util/grub.d/20_linux_xen.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
index 9b1bd7169650..fae1ffe94942 100644
--- a/util/grub.d/20_linux_xen.in
+++ b/util/grub.d/20_linux_xen.in
@@ -216,6 +216,10 @@ while [ "x${xen_list}" != "x" ] ; do
     else
 	xen_loader="multiboot"
 	module_loader="module"
+	if ($grub_file --is-x86-multiboot2 $current_xen); then
+		xen_loader="multiboot2"
+		module_loader="module2"
+	fi
     fi
     while [ "x$list" != "x" ] ; do
 	linux=`version_find_latest $list`
-- 
2.14.3

